ADR/0036-trusted-artifacts.md

```diff
@@ -0,0 +1,54 @@
+# 36. Trusted Artifacts
+
+Date: 2024-06-11
+
+## Status
+
+Accepted
+
+## Context
+
+One of the properties of Konflux is that users should be allowed to include their own Tekton Tasks
+in a build Pipeline, e.g. to execute unit tests, without jeopardizing the integrity of the build
+process. This is distinct from other build systems where a rigid process prevents users from
+applying such customizations. To support this, Konflux build Pipelines use Trusted Artifacts to
+securely share files between Tasks. Enterprise Contract is then responsible for verifying that
+Trusted Artifacts were properly used in the parts of the build Pipeline that affect the build
+outcome, typically the `git-clone`, `prefetch-dependencies`, and `buildah` Tasks.
+
+Trusted Artifacts is inspired by the upcoming work being done by the Tekton Community,
+[TEP-0139](https://github.com/tektoncd/community/blob/main/teps/0139-trusted-artifacts.md). The
+Konflux version is meant to be a stop-gap until that feature is implemented and ready to be used.
+When the time comes, the Konflux implementation should align with what is provided by the Tekton
+Community and this ADR may need a revision.
```

Comment:
Can we state that a new ADR should be created to represent the fact that we are changing as well?

Reply:
Good idea. Done.